final/scripts/app.js

```diff
@@ -64,6 +77,9 @@
     var selected = select.options[select.selectedIndex];
     var key = selected.value;
     var label = selected.textContent;
+    if (!app.selectedCities) {
```

Comment:
Moved this from the startup code. This fixes the edge case I mentioned in our email